### PR TITLE
Changes required for TDI attribuets

### PR DESCRIPTION
**Pull request by saynb**

### CODE REVIEW

File: include/tdi/common/c_frontend/tdi_attributes.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *  @brief Contains TDI Attribute APIs
+ */
+#ifndef _TDI_ATTRIBUTES_H
+#define _TDI_ATTRIBUTES_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+tdi_status_t tdi_attributes_set_value(tdi_attributes_hdl *attributes,
```
Comment (weiqiand) [inline include/tdi/common/c_frontend/tdi_attributes.h:27]: const before the tdi_attributes_hdl

File: src/c_frontend/tdi_table_c.cpp [hunk 2]
```diff
@@ -408,122 +395,23 @@ tdi_status_t tdi_table_action_data_allocate_container_with_fields(
 
 #ifdef _TDI_FROM_BFRT
```
Comment (weiqiand) [inline src/c_frontend/tdi_table_c.cpp:396]: Do we enable the table tdi_table_entry_scope_attributes_allocate now? If we do, please remote the _TDI_FROM_BFRT.

File: src/c_frontend/tdi_table_c.cpp [hunk 3]
```diff
@@ -629,12 +514,11 @@ tdi_status_t tdi_action_id_applicable(const tdi_table_hdl *table_hdl,
 }
 
 #ifdef __TDI_FROM_BFRT
```
Comment (weiqiand) [inline src/c_frontend/tdi_table_c.cpp:516]: If we enable the tdi_table_attributes_set, then we need to remove the __TDI_FROM_BFRT.

File: include/tdi/common/c_frontend/tdi_attributes.h (hunk 1 above)
Comment (saynb) [reply include/tdi/common/c_frontend/tdi_attributes.h:27]: I don't think we should be making the hdl in a setValue function as const

File: src/c_frontend/tdi_table_c.cpp (hunk 2 above)
Comment (saynb) [reply src/c_frontend/tdi_table_c.cpp:396]: No, the c frontend work for attributes is still remaining. We need to add that support once the separation of logic in the CLI from common open source to target specific code is done.

**Review by weiqiand — APPROVED** [review]

**Pull request merged**